mlrun/api/db/sqldb/db.py

```diff
@@ -549,6 +549,12 @@ def list_functions(self, session, name=None, project=None, tag=None, labels=None
         if tag:
             uids = self._resolve_class_tag_uids(session, Function, project, tag, name)
         functions = FunctionList()
+        # the _find_functions bellow searches for functions that contain the tags uids,
+        # and when the uids contains an empty list as a result of a un-existing tag in the database,
+        # it returns all functions because it does not add the uids filter. So, if a tag has been searched
+        # and no corresponding uids have been found, we want to return an empty FunctionList.
+        if tag and not uids:
+            return functions
```

Comment:
I feel like what you did here is giving special handling for the specific case instead of fixing the code to just behave correctly:
When no tag is found, `_find_functions` gets an empty list to the `uids` param, this helper method currently behave the same when `uids` is `None` and when `uids` is an empty list, I believe it's different scenarios (therefore requires different handling), when `uids` is `None`, the default, it's like not provided, therefore the filter shouldn't be applied. If `uids` is anything other than `None` filter should be applied, then if it's empty list, empty function list would be returned.
Although the `_find_functions` helper is only called here currently, it may be called from other places in the future, which gives another reason to have the fix there
Lastly, having the fix there just makes sense and won't require the long comment (since it's not a special handling that requires explanation, it's just the code doing what it makes sense it would)